diff --git a/mrbgems/mruby-bigint/core/bigint.c b/mrbgems/mruby-bigint/core/bigint.c
index 4a561c182..e3811dc6f 100644
--- a/mrbgems/mruby-bigint/core/bigint.c
+++ b/mrbgems/mruby-bigint/core/bigint.c
@@ -1447,7 +1447,7 @@ mrb_bint_as_int64(mrb_state *mrb, mrb_value x)
   uint64_t u = 0;
   size_t len = digits(&m);
 
-  if (len*sizeof(mp_limb) > sizeof(uint64_t)) {
+  if (len*DIG_SIZE > sizeof(uint64_t)*8) {
   out_of_range:
     mrb_raise(mrb, E_RANGE_ERROR, "integer out of range");
   }
@@ -1471,7 +1471,7 @@ mrb_bint_as_uint64(mrb_state *mrb, mrb_value x)
   uint64_t u = 0;
   size_t len = digits(&m);
 
-  if (m.sn < 0 || len*sizeof(mp_limb) > sizeof(uint64_t)) {
+  if (m.sn < 0 || len*DIG_SIZE > sizeof(uint64_t)*8) {
     mrb_raise(mrb, E_RANGE_ERROR, "integer out of range");
   }
   for (size_t i=len-1; ; i--) {
